src/shared/i18n/index.js

```diff
@@ -1,109 +1,46 @@
-import en from 'locales/en';
-import es from 'locales/es';
-import fr from 'locales/fr';
-import ru from 'locales/ru';
-
-// configuration
-const config = {
-  defaultLanguage: 'en',
-  availableLanguages: [
-    {
-      name: 'English',
-      code: 'en',
-      messages: en
-    },
-    {
-      name: 'Deutsch',
-      code: 'de',
-      messages: de
-    },
-    {
-      name: 'Español',
-      code: 'es',
-      messages: es
-    },
-    {
-      name: 'Français',
-      code: 'fr',
-      messages: fr
-    },
-    {
-      name: 'Русский',
-      code: 'ru',
-      messages: ru
-    }
-  ]
-};
-
-// add available language rules from react-intl
-addLocaleData(
-  config.availableLanguages.map(lang =>
-    require(`react-intl/locale-data/${lang.code}`)
-  )
-);
-
-// get translation messages by language code
-const getTranslationsByLangCode = langCode =>
-  config.availableLanguages.find(lang => lang.code === langCode).messages;
-
-/**
- * i18n object
- */
-const i18n = {
-  provider: null,
-  locale: config.defaultLanguage,
-  translations: {},
-  /**
-   * Setup language
-   * @param {object} store
-   */
-  setup(store) {
-    this.locale =
-      getSetting(store.getState(), 'locale') || config.defaultLanguage;
-    this.translations = getTranslationsByLangCode(this.locale);
-
-    this.provider = new IntlProvider(
-      { locale: this.locale, messages: this.translations },
-      {}
-    );
+export const languages = {
+  en: {
+    name: 'English',
+    common: require('locales/en/translation.json')
   },
-  /**
-   * Used for non react files
-   * @param {*} object
-   */
-  formatMessage({ id, defaultMessage, description, values }) {
-    let translation = '';
-    // provider is set
-    if (this.provider) {
-      const { intl } = this.provider.getChildContext();
-      translation = intl.formatMessage(
-        { id, defaultMessage, description },
-        values
-      );
-    } else {
-      // fallback
-      translation = getTranslationsByLangCode(this.locale)[id] || id;
-    }
-
-    return translation;
+  de: {
+    name: 'Deutsch',
+    common: require('locales/de/translation.json')
+  },
+  es: {
+    name: 'Español',
+    common: require('locales/es/translation.json')
   },
-  /**
-   * Return config
-   */
-  getConfig(key) {
-    if (!(key in config)) {
-      throw new Error('i18n config not found.');
-    }
-    return config[key];
+  fr: {
+    name: 'Français',
+    common: require('locales/fr/translation.json')
+  },
+  ru: {
+    name: 'Русский',
+    common: require('locales/ru/translation.json')
   }
 };
 
-// export wrapper for file-manager.js and index.js
-export { IntlProvider };
+const resources = Object.keys(languages).reduce((accumulator, key) => {
+  accumulator[key] = {
+    common: languages[key].common
+  };
+  return accumulator;
+}, {});
+
+i18n.init({
+  fallbackLng: 'en',
+  resources,
+  react: {
+    wait: true
+  },
+  ns: ['common'],
+  defaultNS: 'common',
+  keySeparator: '>',
```

Comment:
Key scanner and ns separator are different here than in the scanner config. I kind of prefer the ones in the scanner. : and .

Reply:
Ok I'll replace them with the scanner separators.